crates/viewer/re_space_view_map/src/map_space_view.rs

```diff
@@ -0,0 +1,334 @@
+use re_space_view::controls::{
+    ASPECT_SCROLL_MODIFIER, HORIZONTAL_SCROLL_MODIFIER, SELECTION_RECT_ZOOM_BUTTON,
+    ZOOM_SCROLL_MODIFIER,
+};
+use re_types::blueprint::{
+    archetypes::MapOptions,
+    components::{Secret, ZoomLevel},
+};
+use re_ui::{ModifiersMarkdown, MouseButtonMarkdown};
+use re_viewer_context::TypedComponentFallbackProvider;
+use re_viewport_blueprint::ViewProperty;
+
+use {
+    egui::{self, Color32, Context},
+    re_log_types::EntityPath,
+    re_space_view::suggest_space_view_for_each_entity,
+    re_types::blueprint::components::MapProvider,
+    re_types::{
+        components::{Color, Radius},
+        SpaceViewClassIdentifier, View,
+    },
+    re_viewer_context::{
+        SpaceViewClass, SpaceViewClassLayoutPriority, SpaceViewClassRegistryError, SpaceViewId,
+        SpaceViewSpawnHeuristics, SpaceViewState, SpaceViewStateExt as _,
+        SpaceViewSystemExecutionError, SpaceViewSystemRegistrator, SystemExecutionOutput,
+        ViewQuery, ViewerContext,
+    },
+    walkers::{HttpTiles, Map, MapMemory, Plugin, Tiles},
+};
+
+use crate::map_visualizer_system::{MapEntry, MapVisualizerSystem};
+use crate::map_windows;
+
+// walkers plugin to visualize points on a Map
+pub struct PositionsOnMap {
+    positions: Vec<MapEntry>,
+}
+
+impl Plugin for PositionsOnMap {
+    fn run(
+        &mut self,
+        _response: &egui::Response,
+        painter: egui::Painter,
+        projector: &walkers::Projector,
+    ) {
+        for entry in &self.positions {
+            // Position of the point we want to put our shapes.
+            let position = entry.position;
+
+            // Project it into the position on the screen.
+            let position = projector.project(position).to_pos2();
+
+            // Radius of the circle
+            let radius = f32::from(
+                *entry
+                    .radii
+                    .unwrap_or(Radius(re_types::datatypes::Float32(10.))),
+            );
+
+            // Color of the circle
+            let color = entry.color.unwrap_or(Color::new(Color32::RED));
+
+            painter.circle_filled(position, radius, color);
+        }
+    }
+}
+
+#[derive(Default)]
+pub struct MapSpaceViewState {
+    tiles: Option<HttpTiles>,
+    map_memory: MapMemory,
+    selected_provider: MapProvider,
+    access_token: String,
```

Comment:
I think `selected_provider` and `access_token` are not really necessary here. We usually try to keep view states as lean as possible. My suggestion:
- remove these two fields
- make `map_memory` an `Option`
- update `map_memory` from `MapViewClass::ui()`

Also, I'd either inline `ensure_and_get_mut_refs` or somehow move it to `MapViewClass`.

Reply:
You're right. Then let's just add a comment above `selected_provider` to indicate that it is used for invalidation.